plugins/SitesManager/API.php

```diff
@@ -1515,20 +1515,46 @@ public function renameGroup($oldGroupName, $newGroupName)
         return true;
     }
 
-    public function getPatternMatchSites($pattern)
+    /**
+     * Find websites matching the given pattern.
+     *
+     * Any website will be returned that matches the pattern in the name, URL or group.
+     * To limit the number of returned sites you can either specify `filter_limit` as usual or `limit` which is
+     * faster.
+     *
+     * @param string $pattern
+     * @param int|false $limit
+     * @return array
+     */
+    public function getPatternMatchSites($pattern, $limit = false)
```

Comment:
Nope it wasn't possible. It would otherwise apply the `filter_limit` later again in the `limit` filter (GenericFilter...) and when `filter_offset` is used no results might be returned. That's why most of the API methods use `$limit` in such a case. Even applying `filter_limit` and `filter_offset` here doesn't work since there might be still a `sort` applied later etc.  

Hopefully in Piwik 4.0 we will rewrite generic filter and apply filter like `sort`, `filter_limit`, `filter_offset` already when fetching the data and not at the end to make all API's much faster.